snapshot-labs/sx-monorepo
# feat: Show proposal timeline on proposal and editor

### Summary

- Extract the same logic as in `Timeline` component (without modal) to a new component `ProposalTimeline` component 
- Reuse new component `ProposalTimeline` in `Timeline` for modal, in `Proposal.vue` for proposal, in `Editor.vue` for the editor
-`ProposalTimeline` will receive `proposal` data from `Timeline` and `Proposal` where as it will receive `space` data from `Editor`
- Format received data to our required data i.e ` network, created, start, min_end, max_end`
- Update `now` every second so that [all time values update real-time ](https://github.com/snapshot-labs/sx-monorepo/issues/71#issuecomment-2270567125)

Closes: https://github.com/snapshot-labs/sx-monorepo/issues/71

### How to test

1. [Go to any proposal ](http://localhost:8080/#/sn:0x05ea5ef0c54c84dc7382629684c6e536c0b06246b3b0981c426b42372e3ef263/proposal/3), timeline modal should be same as before 
2. ![image](https://github.com/user-attachments/assets/db39a312-557e-4177-ae9c-a92b217e8af9)
3. Should display a new timeline in side bar (similar to modal with new design)  
4. ![image](https://github.com/user-attachments/assets/c0c62916-9a67-4d55-8815-341409d6d5b9)
5. In mobile 
6. ![image](https://github.com/user-attachments/assets/23e9f724-2d38-48c3-95d2-c96a4921f673)
7. Off-chain  proposals should have only `End` instead of min and max 
8. ![image](https://github.com/user-attachments/assets/ee10ce96-2dca-440b-8598-b26d6714dded)
9. Go to editor (on-chain and offchain)
10. You should see timeline values according to your space settings
11. ![image](https://github.com/user-attachments/assets/4cb9b067-145e-4770-a265-c634e535ebb8)
12. Off-chain (mobile) 
13. ![image](https://github.com/user-attachments/assets/767e87bb-5541-4e3f-8532-310c2af79632)
